base/math.jl

```diff
@@ -999,6 +999,7 @@ end
 @constprop :aggressive function ^(x::Float64, y::Float64)
     yint = unsafe_trunc(Int, y) # Note, this is actually safe since julia freezes the result
     y == yint && return x^yint
+    y >= 2*inv(eps()) && return x^typemax(Int64)
```

Comment:
To be clear, this is assuming all numbers larger than `2*inv(eps())` are integers and have the same parity as `typemax(Int64)`?

Reply:
Oops. This should be `typemax(Int64)-1`.

Reply:
the reason this is correct is that `inv(eps())` is the point at which `eps(x)=1`, so once you are twice that big, all your floats are even.

Reply:
But why?  How do you know the parity of numbers larger than `maxintfloat`?

Reply:
Oh well, I see `isinteger` gives `true` for numbers larger than `maxintfloat` (basically for the reason you gave above), so I guess that's consistent at least.

Reply:
since the mantissa only has 52 (or fewer) bits, of the exponent is grater than 52, the number is even (and if it's 52, it's an integer)